horace_core/utilities/split_counts.m

```diff
@@ -0,0 +1,111 @@
+function [chunks, idxs] = split_counts(counts, max_counts, varargin)
+%SPLIT_COUNTS_VECTOR Split the given array of counts such that the sum of each
+%  sub-array has a maximum of max_counts or has length 1.
+%
+% The counts counts are not split across bin boundaries - i.e. each sub-array
+% contains only full bins. If a value in counts is greater than max_counts, then
+% that value will comprise its own sub-array.
+%
+% Input:
+% ------
+% counts             A vector of positive integers.
+% max_counts         An integer specifying the maximum sum for each sub-array.
+% total_counts       The sum of all counts values, equal to sum(counts).
+%                    [optional]
+% cumulative_counts  The cumulative sum of the given counts array.
+%                    Should be equal to cumsum(counts). [optional]
+%
+% Output:
+% -------
+% chunks       Cell array of vectors. Concatenation of these arrays will be
+%              equal to a flattened version the inputted counts array, i.e.
+%              counts(:).
+% idxs         The indices at which the input counts array was "split". Has size
+%              [2, n], where n is numel(chunks). Each idxs(:, i) is the
+%              upper and lower index into counts of chunks{i}.
+%
+% Example:
+% --------
+% >> counts = [3, 2, 0, 6, 0, 5, 3, 1, 1, 24, 4, 2, 3, 0];
+% >> max_counts = 11;
+% >> [chunks, idxs] = split_counts(counts, max_counts, sum(counts), cumsum(counts))
```

Comment:
This is a clumsy API: `data, maxsize, [f(g(data)), [g(data)]]` 

The only place this function is used (currently) doesn't use the total and cumulative counts data it calculates for anything else other this function call.

Removing the two optional args will simplify the implementation too as you are guaranteed the calculated values are correct and consistent with the `counts` array argument.

Reply:
If future use-cases need total/cumsum data it could be returned from this function as optional additional return values.